### PR TITLE
PINTS scatter gather

### PR DESCRIPTION
**Pull request by edmundmiller**
Adds a Scatter Gather pattern for PINTS splitting the samples by chromosome to speed up transcript identification.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/nascent/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/nascent _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by jfy133 — APPROVED** [review]
Minor things but otherwise LGTM

File: subworkflows/local/transcript_identification/main.nf [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        SAMTOOLS_MERGE.out.bam.combine(ch_chr),
+        params.assay_type
+    )
+    ch_versions = ch_versions.mix(PINTS_CALLER.out.versions.first())
+
+    // Gather the chromosomes
+    // TODO Tests don't seem to hit this because there's no bidirectional_TREs
+    // Need to collect all of the beds for each chromosome/sample and concatenate them
+    // Nextflow makes this super easy
+    def ch_bidirectional_TREs = PINTS_CALLER.out.unidirectional_TREs.groupTuple(by: [0]).map { meta, beds ->
```
Comment (sateeshperi) [inline subworkflows/local/transcript_identification/main.nf:80]: why use `def` here ?

File: docs/usage.md [hunk 2]
```diff
@@ -77,6 +77,14 @@ The current options for transcript identification include [GroHMM](https://bioco
 
 The default transcript identification option is PINTS, and HOMER if the transcript `assay_type` is `GROseq` but this may change in future releases.
 
+### PINTS
+
+PINTS handles the majority of the transcript identification, since it covers all of the supported assays.
+
+PINTS can use a lot of memory while running, so [a scatter-gather pattern was implemented](github).
```
[the remark this answers is not in the document]
Comment (edmundmiller) [reply docs/usage.md:84]: Thanks for catching that! I was just waiting for a good permalink.

File: subworkflows/local/transcript_identification/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// TODO Test ChrY filtering
```
[the remark this answers is not in the document]
Comment (edmundmiller) [reply subworkflows/local/transcript_identification/tests/main.nf.test:1]: Yeah, kind of. This is more of a note for myself or anyone working on it in the future if there is an issue with filtering the chromosomes. https://github.com/nf-core/nascent/blob/136a9ca2390121639e823e39e508afe9b6970d77/subworkflows/local/transcript_identification/main.nf#L49-L53

It is more about testing a fasta with some of those values or refactoring it to allow user input if they want to skip any chromosomes. It's a feature I think is cool, but no one is asking for it, and I have a dissertation to write up. 🫠

File: subworkflows/local/transcript_identification/main.nf (hunk 1 above)
Comment (edmundmiller) [reply subworkflows/local/transcript_identification/main.nf:80]: The new extension told me to do it 🤷🏻‍♂️ If you turn on the future warnings, I think?

**Comment by edmundmiller** [comment]
Thanks, @jfy133 and @sateeshperi, for giving it a look!

**Pull request merged**